include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -607,17 +600,27 @@ namespace internal
       this->is_src_locally_owned = is_src_locally_owned;
 
       const auto targets_with_indexset = process.get_requesters();
+      std::vector<types::global_dof_index> ghost_indices;
+
+#ifndef DEAL_II_WITH_MPI
+      Assert(targets_with_indexset.empty(), ExcInternalError());
+#else
 
       std::map<unsigned int, std::vector<types::global_dof_index>>
                                indices_to_be_sent;
       std::vector<MPI_Request> requests;
       requests.reserve(targets_with_indexset.size());
+      const unsigned int my_rank =
+        Utilities::MPI::this_mpi_process(communicator);
 
       {
         std::vector<types::global_dof_index> indices;
 
         for (const auto &i : targets_with_indexset)
           {
+            if (i.first == my_rank)
+              continue;
```

Comment:
It looks like this change causes the code to hang. It means the code depends on having an MPI rank sending MPI messages to itself. Not the only problematic performance here, as observed in #14968. I will look into this later.

Reply:
OK, addressed, I simply had a wrong condition further down.